src/services/gridPageFetcher.ts

```diff
@@ -1,116 +1,114 @@
-import { DecisionWrapper } from "../models/verification";
+import { SubjectWrapper } from "../models/subject";
 import { VerificationParser } from "./verificationParser";
 
-export type PageFetcher = (elapsedItems: number) => Promise<DecisionWrapper[]>;
+/**
+ * A context object that is passed to the page fetcher function after every call
+ * this can be used to keep track of state such as the current page, where the
+ * page fetcher would increment the page number after every call.
+ */
+export type PageFetcherContext = Record<string, unknown>;
+export type PageFetcher<T extends PageFetcherContext = any> = (context: T) =>
+    Promise<SubjectWrapper[]>;
```

Comment:
we should allow for immutable context. We store their last context and pass it back to them on next invocation.

```suggestion
    Promise<{ subjects: SubjectWrapper[], context: T}>;
```